### DISCUSSION TITLE
form with bind field

### DISCUSSION THREAD

**Discussion by ionv06** [sqlpage/SQLPage#614] [Q&A]
Hi,
How can I bind an options field from a form to a text field? 
```sql
select
    'form' as component,
    'Invoices' as title,
    '' as validate;
select
    'CIF' as label,
    'cif' as name,
    'select' as type,
    (select 
        JSON_ARRAYAGG(JSON_OBJECT(
             'value', cif,
             'label', name
         )) 
         from companies
    ) as options;
```
I want to add another field in the form, 'name', whose value should be linked to the value selected in the 'cif' field defined above; if the user selects another value from the 'cif' dropdown list, the value of this field must change ('cif' is a company id and 'name' is, of course, the company name)

**Comment by DSMejantel** [comment]
Hi,
You can try with thé multistep form like in the examples from the documentation.

**Comment by lovasoa** [comment]
Hi ! 
  Your sql pages are entirely rendered on the server side, which prevents dynamic client-side behavior. However, you can easily use a multi-step form to achieve a similar behavior.
  
  You can have a look at [/examples/multistep-form/index.sql](https://github.com/lovasoa/SQLpage/blob/main/examples/official-site/examples/multistep-form/index.sql) which is heavily commented.

**Comment by YunusMechery** [comment]
@ionv06 try: https://github.com/lovasoa/SQLpage/discussions/628
